demo/src/main/java/com/novoda/movies/rateable/RateableMoviesActivity.java

```diff
@@ -0,0 +1,65 @@
+package com.novoda.movies.rateable;
+
+import android.os.Bundle;
+import android.support.annotation.Nullable;
+import android.support.v7.app.AppCompatActivity;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.widget.Toast;
+
+import com.novoda.movies.R;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class RateableMoviesActivity extends AppCompatActivity {
+
+    private final FakeMovieRepository fakeMovieRepository = new FakeMovieRepository();
+    private final RateableMoviesAdapter rateableMoviesAdapter = new RateableMoviesAdapter();
+
+    @Override
+    protected void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_rateable_movies);
+
+        RecyclerView recyclerView = findViewById(R.id.rateable_movies_recycler_view);
+        recyclerView.setLayoutManager(new LinearLayoutManager(this));
+        recyclerView.setAdapter(rateableMoviesAdapter);
+
+        fakeMovieRepository.observeChanges(observer);
```

Comment:
but it does seem to be trendy nowadays to have subscribe to an observable and receive periodic updates.

here, we'll get update of `List<Movie>` whenever we subscribe. Also if we use the `FakeMovieRepository.onRate(long id, float rating)` or `FakeMovieRepository.onToggleLike(long id)` to react to a user action, we'll get another update with the action applied.

from the `list<Movie>`, we convert it into the viewmodel that we bind to our viewholder.